load_balance.go

```diff
@@ -358,21 +376,39 @@ func refreshLoadInfo(li *ClusterLoadInfo) error {
 			newHostPairs[host] = publicIP
 			tk := cloud + "." + region + "." + zone
 			tk_star := cloud + "." + region // Used for topology_keys of type: cloud.region.*
-			hosts, ok := newZoneList[tk]
-			if !ok {
-				hosts = make([]string, 0)
-			}
-			hosts_star, ok_star := newZoneList[tk_star]
-			if !ok_star {
-				hosts_star = make([]string, 0)
+			if nodeType == "primary" {
```

Comment:
Both the `if` and `else` blocks look similar. Please see if the common logic can be put in a method.